core/services/relay/evm/chain_reader.go

```diff
@@ -155,11 +176,32 @@ type ChainReaderService interface {
 type chainReader struct {
 	lggr logger.Logger
 	lp   logpoller.LogPoller
+	// key being contract name
+	chainContractReaders map[string]types.ChainContractReader
+	chain                evm.Chain
 }
 
-// chainReader constructor
-func NewChainReaderService(lggr logger.Logger, lp logpoller.LogPoller) (*chainReader, error) {
-	return &chainReader{lggr.Named("ChainReader"), lp}, nil
+func registerLogPollerFilters(cr chainReader, contractAddress common.Address, contractName string) error {
+	chainContractReader, exists := cr.chainContractReaders[contractName]
+	if !exists {
+		return fmt.Errorf("%s contract is not defined in chain reader config", contractName)
+	}
+
+	abiEvents := chainContractReader.ParsedContractABI.Events
+	for _, crd := range chainContractReader.ChainReaderDefinitions {
+		if crd.ReadType == types.Event {
+			eventSig := abiEvents[crd.ChainSpecificName].ID
+			err := cr.lp.RegisterFilter(logpoller.Filter{
+				Name:      crd.ChainSpecificName,
+				EventSigs: evmtypes.HashArray{eventSig},
+				Addresses: evmtypes.AddressArray{contractAddress},
+			})
```

Comment:
Another thing we should probably include in the RelayConfig is a `startingBlock`.  Eventually, RegisterFilter should support accepting that, but for now the way most products do it is to call `lp.RegisterFilter` and then `lp.Replay(startingBlock)` immediately after that.

Reply:
(We could choose to skip this for the present POC, but would be good to have if it's not too difficult to include.)

Reply:
Its very easy to add. I am just wondering how Replay should be used here. Seems like we need to set one value for the entirety of chainReader, does it make sense to do it this way?